Add post for Rust 1.69.0 (#1097)
rust-lang/blog.rust-lang.org: posts/2023-04-20-Rust-1.69.0.md

---
layout: post
title: "Announcing Rust 1.69.0"
author: The Rust Release Team
release: true
---

> The Rust team is happy to announce a nice version of Rust, 1.69.0. Rust is a programming language empowering everyone to build reliable and efficient software.

Reviewer: ```suggestion
The Rust team is happy to announce a new version of Rust, 1.69.0. Rust is a nice programming language empowering everyone to build reliable and efficient software.
```

Alternative idea.

Author: Kinda prefer the current one, the nice version is 1.69.0.

If you have a previous version of Rust installed via rustup, you can get 1.69.0 with:

```console
rustup update stable
```

If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.69.0](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1690-2023-04-20) on GitHub.

If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!

## What's in 1.69.0 stable

Rust 1.69.0 doesn't contain major new features. Instead, it focuses on smaller improvements.

### Cargo now suggests to automatically fix some warnings

> Rust 1.29.0 added a Cargo subcommand to automatically fixing some compilation warnings. Since then, the amount of warnings that can be fixed automatically steadily increased, and support for automatically fixing a subset of Clippy warnings was introduced.

Reviewer 2: ```suggestion
Rust 1.29.0 added a Cargo subcommand to automatically fix some compilation warnings. Since then, the amount of warnings that can be fixed automatically has steadily increased, and support for automatically fixing a subset of Clippy warnings was introduced.
```

> Rust 1.29.0 added a Cargo subcommand to automatically fix some compilation warnings. Since then, the amount of warnings that can be fixed automatically has steadily increased, and support for automatically fixing a subset of Clippy warnings was introduced.

Reviewer 3: ```suggestion
Rust 1.29.0 added the `cargo fix` subcommand to automatically fix some simple compiler warnings. Since then, the number of warnings that can be fixed automatically continues to steadily increase. In addition, support for automatically fixing some simple Clippy warnings has also been added.
```

> Starting from Rust 1.69.0, Cargo will suggest running `cargo fix` or `cargo clippy --fix` when it detects some warnings are automatically fixable.

Reviewer 3: ```suggestion
In order to draw more attention to these increased capabilities, Cargo will now suggest running `cargo fix` or `cargo clippy --fix` when it detects warnings that are automatically fixable:

    warning: unused import: `std::hash::Hash`
     --> src/main.rs:1:5
      |
    1 | use std::hash::Hash;
      |     ^^^^^^^^^^^^^^^
      |
      = note: `#[warn(unused_imports)]` on by default
    
    warning: `foo` (bin "foo") generated 1 warning (run `cargo fix --bin "foo"` to apply 1 suggestion)

Note that the full Cargo invocation shown above is only necessary if you want to precisely apply fixes to a single crate. If you want to apply fixes to all the default members of a workspace, then a simple `cargo fix` (with no additional arguments) will suffice.
```

### Debug information is not included in build scripts by default anymore

To improve compilation speed, Cargo now avoids emitting debug information in build scripts by default. There will be no visible effect when build scripts execute successfully, but backtraces in build scripts will contain less information.

> If you need to debug a buggy build script, you can add this snippet to your `Cargo.toml` to emit debug information again:

Reviewer 2: ```suggestion
If you want to debug a build script, you can add this snippet to your `Cargo.toml` to emit debug information again:
```

during debugging one might discover that it's not the build script that has the bug but a component it calls, so removing the word "buggy" is better IMO.
